commands/serve.go

```diff
@@ -88,6 +83,18 @@ func (r *serveRunner) run(_ *cobra.Command, _ []string) error {
 	return nil
 }
 
+func (r *serveRunner) CreateServer() *http.Server {
```

Comment:
I'd say we can keep it private until needed.
```suggestion
func (r *serveRunner) createServer() *http.Server {
```